packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1998,9 +1999,7 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
       yield* entry.route.overlayEntries;
   }
 
-  List<OverlayEntry> _lastUpdatedOverlayEntries = const <OverlayEntry>[];
-
-  void _flushHistoryUpdates() {
+  void _flushHistoryUpdates({bool rearrangeOverlay = true}) {
```

Comment:
Add an assert to the "_RouteLifecycle.add" and "_RouteLifecycle.push/_RouteLifecycle.pushReplace/_RouteLifecycle.replace" case that rearrangeOverlay is true? (If it is false in those cases the new route would not show up).